### ISSUE TITLE
Nuget packages FileNotFoundException

### ISSUE THREAD

**Issue by jcweaver997** [godotengine/godot#53693]
### Godot version

v4.0.dev.mono.custom_build [f9aec342d]

### System information

Windows 10

### Issue description

After installing a nuget package, I received the following error:

```
E 0:00:01:0321   GDMonoUtils::debug_send_unhandled_exception_error: System.IO.FileNotFoundException: Could not load file or assembly 'MessagePack, Version=2.3.0.0, Culture=neutral, PublicKeyToken=b4a0369545f0a1be' or one of its dependencies.
  <C++ Error>    Unhandled exception
  <C++ Source>   modules\mono\mono_gd\gd_mono_utils.cpp:450 @ GDMonoUtils::debug_send_unhandled_exception_error()

```
As far as I can tell, this happens with all nuget packages. I tested referencing a dll, and it worked fine, just the nuget packages that have issues.

mono logs
[2021-10-11_23.16.29_20640.log](https://github.com/godotengine/godot/files/7326663/2021-10-11_23.16.29_20640.log)

### Steps to reproduce

1. Create a new godot mono project
2. Create a scene
3. Add a new script to the root Node
4. Open the C# project (I used rider, but same issue in vs2019)
5. Add the MessagePack nuget package
6. Add `MessagePackSerializer.Typeless.Serialize(123);` to the _Ready of the script created
7. Play the scene
8. You should get the same error

### Minimal reproduction project

The following project follows the steps above
[zxc.zip](https://github.com/godotengine/godot/files/7326657/zxc.zip)

**Comment by steelywing** [comment]
I found Godot 3.4 also doesn't copy the NuGet DLL file to mono output
- If the NuGet package is local package (using `dotnet add package XXX -s XXX`)
  - Because I need port it to .NET standard, the package work with .NET Core only
- Or the NuGet package (Online) is included in other reference project (not the Godot project)
- If I manually copy the DLL to `.mono\temp\bin\Debug`, it works

I have wasted a whole day on it, at first I think it is .NET / NuGet package version issue.

**Comment by raulsntos** [comment]
This sounds like a duplicate of #42271

**Comment by MikeSchulze** [comment]
> <PropertyGroup>
>   <CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies>
> </PropertyGroup>

Thanks i run into same probem this helps me a lot

**Comment by aignermax** [comment]
still running into this error at Godot 4.1.3 -> the <CopyLocalLockFileAssemblies>true Solution from above kind of fixes that issue, but shouldn't it work out of the box?

**Comment by raulsntos** [comment]
I'm unable to reproduce this issue with Godot 4.2.1 using the provided MRP after making the necessary changes to be compatible with the latest version of Godot.

@aignermax Please upload a [minimal reproduction project](https://github.com/godotengine/godot/blob/4c1f11944ebfe1204f21c6ac488c7a2294e1d520/CONTRIBUTING.md#provide-a-simple-example-project) if you can still reproduce this.

**Comment by aignermax** [comment]
ok so here is the current 4.2.1 version of godot trying to use the Mathnet.dll
Visual Studio says all is fine, but Godot doesn't
![image](https://github.com/godotengine/godot/assets/18228325/2b421df3-ca8e-467a-bb4f-aa897a40ca4b)

but as soon as you add the <CopyLocalLockFileAssemblies>true into the Propertygroup of the .CsProj file, then suddenly it recognizes all of the nuget packages properly 

![image](https://github.com/godotengine/godot/assets/18228325/9766c898-b401-4b49-9316-e2663d8683c5)

here is the code - simply edit the .csproj file and hit f5 in the godot editor - no need to restart it even. 
you can see that using a nuget package only works when you have CopyLocalLockFileAssemblies set to true. 


here the MRP
https://github.com/aignermax/GodotNugetBug/tree/main

**Comment by raulsntos** [comment]
@aignermax Your MRP has the line `<EnableDynamicLoading>true</EnableDynamicLoading>` commented. Godot 4.0+ requires enabling dynamic loading.

When testing the MRP with dynamic loading enabled I was unable to reproduce the issue. No need to use `<CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies>`.

**Comment by aignermax** [comment]
you are right, my bad, sorry for confusing those two, so I can confirm that it works now without the CopyLocalLockFileAssemblies tag. [Simley of a cute dog that wants to be loved, but he has destroyed the whole flat while you were absent]
![DALL·E 2024-01-11 10 54 27 - A cartoon-style image of a cute, guilty-looking dog with big, expressive eyes  The dog is sitting in the middle of a messy, destroyed living room, sur](https://github.com/godotengine/godot/assets/18228325/6111075d-2ffe-4b81-a940-46a6b86347cf)
